pkg/apis/cluster/v1alpha1/machineset_types.go

```diff
@@ -70,6 +75,24 @@ type MachineSetSpec struct {
 	Template MachineTemplateSpec `json:"template,omitempty"`
 }
 
+type MachineSetDeletePolicy string
+
+const (
+	// SimpleMachineSetDeletePolicy prioritizes both Machines that have the annotation
+	// "machineset.clusters.k8s.io/delete-me=yes" and Machines that are unhealthy
+	// (Status.ErrorReason or Status.ErrorMessage are set to a non-empty value).
```

Comment:
Wouldn't it make sense to prioritize the annotated and unhealthy machines in all cases? If we do that, then is there really a use case for the `Simple` policy rather than just having `Oldest` and `Newest`?

Reply:
I agree. In that case, we'll need to decide on a new default policy to replace `Simple`. I think `Newest` would make a sane default. When scaling up and down your cluster it makes more sense to churn through the same machines.

Reply:
I'm a bit torn. `Oldest` would help ensure that users aren't relying on workloads being sticky as much, while also allowing for clusters to adopt newer base OS images which could include security errata faster. At the same time `Newest` would potentially allow for less workload churn.

@roberthbailey thoughts?

Reply:
> `Oldest` would help ensure that users aren't relying on workloads being sticky as much

I just wonder if it's up to the MachineSet to police how users are running their workloads.

> while also allowing for clusters to adopt newer base OS images which could include security errata faster.

I get this point; however, in the event of a new image rollout, wouldn't you rather roll it out in a controlled manner vs. relying on the MachineSet to eventually cycle out old machines?
I definitely see value in `Oldest` with regard to the use cases you bring up, but as far as defaults go, I think `Newest` makes more sense due to the impact of workload churn.

Reply:
Sounds good to me. I would like to get some feedback from @roberthbailey or @justinsb for thoughts on how GKE and/or KOPS approaches this problem.

Reply:
Hey @detiber , just wanted to check in on this. Do you think it'd be worthwhile moving this discussion to the agenda of next week's cluster-api meeting? I'm afraid that this will get stale and/or forgotten.

Reply:
@erstaples I think adding it to next week's agenda is a great idea